src/test/ui/typeck/issue-87181/enum-variant.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0599]: no method named `foo` found for fn item `fn() -> Foo {Foo::Tup}` in the current scope
+  --> $DIR/enum-variant.rs:14:15
+   |
+LL |     thing.bar.foo();
+   |               ^^^ method not found in `fn() -> Foo {Foo::Tup}`
+   |
+   = help: `thing.bar` is the constructor of of an enum variant
```

Comment:
Looking at these, I think we can change the output (and remove the need for the new `fn expr_to_snippet`) by changing it to be

```suggestion
LL |     thing.bar.foo();
   |     --------- ^^^ method not found in `fn() -> Foo {Foo::Tup}`
   |     |
   |     this is the constructor of of an enum variant
```

Changing it that way also neatly sidesteps the problem of "the user wrote `thing\n.bar` or the expression is super long and now the text is hard to read".

Reply:
I removed the snippet to yield this:

```
LL |     thing.bar.foo();
   |               ^^^ method not found in `fn() -> Foo {Foo}`
   |
   = help: this is the constructor of of a struct
```

But I'm not sure how I'd make it _exactly_ like you showed, i.e. with `help` removed and the extra `---`

Reply:
Oh I just noticed the double "of" in the message, I will address that now as well

Reply:
I would change it to be `span_label(expr.span, ...` to make it point at `thing.bar` so that it is clearer what we're talking about.